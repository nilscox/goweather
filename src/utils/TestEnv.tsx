import * as React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import store from '../store';

type TestEnvProps = {
  children: React.ReactNode;
};

const TestEnv = ({ children }: TestEnvProps) => (
  <Provider store={store}>
    <Router>
      { children }
    </Router>
  </Provider>
);

export default TestEnv;
