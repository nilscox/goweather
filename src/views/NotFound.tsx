import * as React from 'react';
import { Container } from 'reactstrap';

import { PageTitle } from '../components';

/**
 * Not found page. Displays an error message when the URL does not match
 * any route defined by the router.
 */
const NotFound = () => (
  <Container className="pb-4">
    <PageTitle>404 - Page not found.</PageTitle>
  </Container>
);

export default NotFound;
