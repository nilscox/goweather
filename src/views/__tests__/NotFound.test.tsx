import * as React from 'react';
import * as renderer from 'react-test-renderer';

import TestEnv from '../../utils/TestEnv';

import NotFound from '../NotFound';

test('NotFound', () => {
  const tree = renderer
    .create(<TestEnv><NotFound /></TestEnv>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
