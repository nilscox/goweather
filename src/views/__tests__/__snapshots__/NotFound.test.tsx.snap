// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound 1`] = `
<div
  className="pb-4 container"
>
  <h1
    className="text-center py-4 css-1aylpg"
  >
    404 - Page not found.
  </h1>
</div>
`;
